core/src/main/java/org/apache/accumulo/core/conf/Property.java

```diff
@@ -429,15 +429,34 @@ public enum Property {
   MANAGER_FATE_METRICS_MIN_UPDATE_INTERVAL("manager.fate.metrics.min.update.interval", "60s",
       PropertyType.TIMEDURATION, "Limit calls from metric sinks to zookeeper to update interval.",
       "1.9.3"),
-  MANAGER_FATE_THREADPOOL_SIZE("manager.fate.threadpool.size", "64", PropertyType.COUNT,
-      "The number of threads used to run fault-tolerant executions (FATE)."
-          + " These are primarily table operations like merge.",
-      "1.4.3"),
+  MANAGER_USER_FATE_CONFIG("manager.user.fate.config", "{"
+      + "\"TABLE_CREATE,TABLE_DELETE,TABLE_RENAME,TABLE_ONLINE,TABLE_OFFLINE,NAMESPACE_CREATE,NAMESPACE_DELETE,NAMESPACE_RENAME,TABLE_TABLET_AVAILABILITY,SHUTDOWN_TSERVER\": 1,"
+      + "\"TABLE_BULK_IMPORT2\": 2,"
+      + "\"TABLE_COMPACT,TABLE_CANCEL_COMPACT,COMMIT_COMPACTION\": 4,"
+      + "\"TABLE_MERGE,TABLE_DELETE_RANGE,TABLE_SPLIT,SYSTEM_SPLIT,TABLE_CLONE,TABLE_IMPORT,TABLE_EXPORT\": 2"
+      + "}", PropertyType.USER_FATE_CONFIG,
+      "The number of threads used to run user-initiated fault-tolerant "
+          + "executions (FATE). These are primarily table operations like merge. Each key/value "
+          + "of the provided JSON corresponds to one thread pool. Each key is a list of one or "
+          + "more FATE operations and each value is the number of threads that will be assigned "
+          + "to the pool.",
+      "4.0.0"),
+  MANAGER_META_FATE_CONFIG("manager.meta.fate.config",
+      "{\"TABLE_COMPACT,TABLE_CANCEL_COMPACT,COMMIT_COMPACTION\": 4,"
+          + "\"TABLE_MERGE,TABLE_DELETE_RANGE,TABLE_SPLIT,SYSTEM_SPLIT\": 2}",
+      PropertyType.META_FATE_CONFIG,
+      "The number of threads used to run system-initiated fault-tolerant "
+          + "executions (FATE). These are primarily table operations like merge. Each key/value "
+          + "of the provided JSON corresponds to one thread pool. Each key is a list of one or "
+          + "more FATE operations and each value is the number of threads that will be assigned "
+          + "to the pool.",
+      "4.0.0"),
```

Comment:
Several comments here:
- This is my best attempt as to what fate ops are valid META fate ops. I may have excluded some that should have been included or vice-versa.
- What is the proper way to deprecate the old MANAGER_FATE_THREADPOOL_SIZE property
- These default pool sizes and default pools probably aren't optimal. Input into how the FATE ops should be split and how many threads for each pool would be appreciated. This was just my best attempt

Reply:
I marked as deprecated in 4.0 and log a warning if the property is set mentioning the replacement props.